pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -377,6 +425,14 @@ func (e *EndpointController) syncService(key string) {
 	}
 }
 
+func verifyPodHostNamesAreEqual(newPodHostNames string, oldAnnotations map[string]string) bool {
+	oldPodHostNames := ""
+	if oldAnnotations != nil {
+		oldPodHostNames = oldAnnotations[endpoints.PodHostNamesAnnotation]
+	}
+	return oldPodHostNames == newPodHostNames
```

Comment:
this is safe as long as Go is used for all producers - go's marshaller sorts map input.  You might not want to count on that, or else at least document it and that it is safe, albeit wasteful, to get this wrong